core/templates/services/math-interactions.service.spec.ts

```diff
@@ -279,4 +279,72 @@ describe('MathInteractionsService', () => {
     expect(mathInteractionsService.replaceAbsSymbolWithText(
       '||')).toBe('abs()');
   });
+
+  it('should get terms from given expression', function() {
+    // Split by addition.
+    expect(mathInteractionsService.getTerms('3+4*a')).toEqual(
+      ['3', '4*a']);
+    expect(mathInteractionsService.getTerms('4-(-beta)')).toEqual(
+      ['4', '-((-beta))']);
+    expect(mathInteractionsService.getTerms('3*10^(-1)')).toEqual(
+      ['3*10^(-1)']);
+    expect(mathInteractionsService.getTerms('a-x+4.5')).toEqual(
+      ['a', '-(x)', '4.5']);
+    expect(mathInteractionsService.getTerms('4----5')).toEqual(
+      ['4', '-(---5)']);
+    expect(mathInteractionsService.getTerms('100 + 20 + 3')).toEqual(
+      ['100', '20', '3']);
+    expect(mathInteractionsService.getTerms('4-sqrt(x + alpha)')).toEqual(
+      ['4', '-(sqrt(x+alpha))']);
+    expect(mathInteractionsService.getTerms('a^2+b^2+2*a*b')).toEqual(
+      ['a^2', 'b^2', '2*a*b']);
+    expect(mathInteractionsService.getTerms('pi/(4+3)')).toEqual(
+      ['pi/(4+3)']);
+    expect(mathInteractionsService.getTerms('tan(30)-(-cos(60))')).toEqual(
+      ['tan(30)', '-((-cos(60)))']);
+
+    // Split by multiplication.
+    expect(mathInteractionsService.getTerms('4*a', false)).toEqual(
```

Comment:
That wouldn't happen since split by multiplication is to be done after split by addition. So if an expression is `a+b*2`, it would be first split into `[a, b*2]` and then to match individual terms we will split each term by multiplication and division so `b*2` will be split into `[b, 2]`. If the expression was `(a+b)*2` then it would be considered as a single term and while matching, it would be split into `[a+b, 2]`. Does that make things clear?